group/templates/group/profile.html

```diff
@@ -39,19 +41,24 @@ <h1>{{ group.name }}</h1>
                 {% endif %}
             {% elif group.join_mode == "REQUEST"%}
                 {% if request_exists %}
-                    <button class="btn btn-secondary join_button request" data-group-id="{{group.id}}">Withdraw Request</button>
+                    <button class="btn btn-secondary join_button request" data-group-id="{{group.id}}" data-is-authenticated="{{ user.is_authenticated }}">Withdraw Request</button>
                 {% else %}
-                    <button class="btn btn-secondary join_button request" data-group-id="{{group.id}}">Request to Join</button>
+                    <button class="btn btn-secondary join_button request" data-group-id="{{group.id}}" data-is-authenticated="{{ user.is_authenticated }}">Request to Join</button>
                 {% endif %}
             {% else %}
-                <button class="btn btn-secondary join_button" data-group-id="{{group.id}}">Join Group</button>
+                <button class="btn btn-secondary join_button" data-group-id="{{group.id}}" data-is-authenticated="{{ user.is_authenticated }}">Join Group</button>
             {% endif %}
         {% else %}
             {% if group.join_mode == "REQUEST"%}
-            <button class="btn btn-secondary join_button request" data-group-id="{{group.id}}">Request to Join</button>
+            <button class="btn btn-secondary join_button request" data-group-id="{{group.id}}" data-is-authenticated="{{ user.is_authenticated }}">Request to Join</button>
             {% else %}
-            <button class="btn btn-secondary join_button" data-group-id="{{group.id}}">Join Group</button>
+            <button class="btn btn-secondary join_button" data-group-id="{{group.id}}" data-is-authenticated="{{ user.is_authenticated }}">Join Group</button>
```

Comment:
repeating ```data-is-authenticated="{{ user.is_authenticated }} ```
this all of the button seems inefficient now I see it. you can leave it for now but I think we should refactor this part. 
I will look into it.